src/main/java/uk/gov/hmcts/reform/migration/CaseMigrationProcessor.java

```diff
@@ -226,6 +227,103 @@ public void migrateQuery(EsQuery query) {
         }
     }
 
+    @SneakyThrows
+    public void migrateQueryByBatch(EsQuery query, String searchAfter, int batchSize) {
+        log.info("Batch mode on!");
+        requireNonNull(query);
+        requireNonNull(caseType);
+        requireNonNull(migrationId);
+
+        if (batchSize <= 0) {
+            throw new IllegalArgumentException("batchSize must be greater than 0");
+        }
+
+        String userToken =  idamRepository.generateUserToken();
+        if (isBlank(searchAfter)) {
```

Comment:
Is searchAfter the number of the batch we're going to try to run? Do we know what the max number for this should be for a given batch?